### DISCUSSION TITLE
posts/csharpier/

### DISCUSSION THREAD

**Discussion by giscus** [BYJRK/byjrk.github.io#7] [Announcements]
# posts/csharpier/

# 简介 CSharpier 是一个针对 C# 代码的格式化工具，它可以帮助开发者自动化地调整代码的格式，使其更加一致和易于阅读。CSharpier 提供了丰富的配置选项

https://blog.coldwind.top/posts/csharpier/

**Comment by wsgaero** [comment]
首先辛苦up主了，有个关于接口特性和用法的一个问题：测试发现，如果不对接口泛型对象进行约束，也可以直接用接口的静态方法。

```c#
public interface IDeserializer<T>
{
    static T? Deserialize(string value) => JsonSerializer.Deserialize<T>(value);
}

class Student
{
    public string Name { get; set; }
    public int Age { get; set; }

    public override string ToString() => $"Name: {Name}, Age: {Age}";
}
```

下面直接调用接口的静态方法进行反序列化，也是可以的。up主，两种定义方式，有何区别呢？

```c#
var student = IDeserializer<Student>.Deserialize("{\"Name\":\"John Doe\",\"Age\":42}");
Debug.WriteLine(student);
```

**Reply by BYJRK** [reply]
你下面的代码本身就为泛型接口提供了 `Student` 这个类型，所以接口的 `Deserialize` 方法就知道了它的返回值，以及它调用 `JsonSerializer.Deserialize<T>` 方法时该提供什么类型。我没太明白你的问题，以及你说的两种定义方式，另一种指的是什么。

另外，你为啥要在这个文章的评论区提出这个问题？😂

**Comment by Xnjikol** [comment]
安装之后vs特别卡，up主有碰到这样的问题吗？每次打开一个没打开过的代码文件会卡5-7秒没有响应，打开后vs会提示CSharpier影响了性能，第二次打开这个文件就不会卡了。

**Reply by BYJRK** [reply]
那个文件很大吗？比如大几千上万行的

**Reply by Xnjikol** [reply]
不是很大，我自己写的一般就几百行，禁用CSarpier插件后确实不卡了，不清楚为什么，不过格式化效果挺好，现在先暂时这样用着了

**Reply by BYJRK** [reply]
你的csharpier是最新版的吗？还有你的电脑配置如何？我基本上没见过它卡过。之前用CodeMaid，那玩意是真的卡，代码如果行数比较多，我根本不敢开它

**Reply by Xnjikol** [reply]
哦我也装了codemaid，而且一直放着没管它了，可能是这俩冲突了，我把codemaid卸载了，目前没有卡顿了，谢谢up主，不过 不知道为什么vs总是提示是CSharpier卡。（codemaid插件很早就装了，搞忘了）

**Comment by Lj-dog** [comment]
这资料好少，Bing搜出来第一条就是这。 请问up 这有快捷键吗 或者像CodeMaidresharp那样保存自动reformat的功能？ 感谢

**Reply by Chendaqian** [reply]
目前应该没有自动整理功能
